source/deploying-with-kubernetes/kubernetes-conf.rst

```diff
@@ -0,0 +1,320 @@
+.. Copyright (C) 2018 Wazuh, Inc.
+
+.. _kubernetes_conf:
+
+Kubernetes Installation & Configuration
+=========================================
+
+- `Installation`_
+- `Pre-requisites`_
+- `Overview`_
+- `Verifying the deployment`_
+- `Agents`_
+
+
+Installation
+------------
+
+To select the best installation type for you, go to the `Official installation guide. <https://kubernetes.io/docs/setup/>`_
+
+Pre-requisites
+--------------
+
+
+    - Kubernetes cluster already deployed.
+
+    - Kubernetes can run on a wide range of Cloud providers and bare-metal environments, this repository focuses on AWS. It was tested using Amazon EKS. You should be able to:
+
+        - Create Persistent Volumes on top of AWS EBS when using a *volumeClaimTemplates*
+        - Create a record set in AWS Route 53 from a Kubernetes LoadBalancer.
+
+    - Having at least two Kubernetes nodes in order to meet the *podAntiAffinity* policy.
+
+Overview
+--------
+
+StateFulSet and Deployments Controllers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Like a *Deployment*, a *StatefulSet* manages Pods that are based on an identical container specification, but it maintains an identity attached to each of its pods. These pods are created from the same specification, but they are not interchangeable: each one has a persistent identifier maintained across any rescheduling.
+
+It is useful for stateful applications like databases that save the data to a persistent storage. The states of each Wazuh manager as well as Elasticsearch are desirable to maintain, so we declare them using StatefulSet to ensure that they maintain their states in every startup.
+
+Deployments are intended for stateless use and are quite lightweight and seem to be appropriate for Logstash, Kibana and Nginx, where it is not necessary to maintain the states.
+
+Users can read more information about the *Persistent volumes* in the `Kubernetes Official documentation. <https://kubernetes.io/docs/concepts/storage/persistent-volumes/>`_
```

Comment:
We could comment more about persistent volumes and their use. Add a brief explanation and comment that they are used to store data of interest from both Wazuh and Elasticsearch. I don't know if in this section or open a subsection later explaining it. The idea is that they work in a similar way to docker volumes (vital for updates).